docs/aws/audit/lambdamonitoring/rules/calls_in_cloudtrail.mdx

```diff
@@ -24,6 +24,228 @@ AWSWAF, PCIDSS
+
+By following these steps, you can ensure that CloudTrail is properly configured to log events for AWS Lambda, thereby preventing the misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To ensure that AWS CloudTrail is enabled for AWS Lambda using the AWS CLI, follow these steps:
+
+1. **Create a CloudTrail Trail:**
+   First, create a CloudTrail trail if you don't already have one. This trail will log all the events, including those from AWS Lambda.
+
+   ```sh
+   aws cloudtrail create-trail --name my-trail --s3-bucket-name my-trail-bucket
+   ```
+
+2. **Start Logging:**
+   Enable logging for the created CloudTrail trail to ensure that it starts capturing events.
+
+   ```sh
+   aws cloudtrail start-logging --name my-trail
+   ```
+
+3. **Verify CloudTrail Status:**
+   Check the status of the CloudTrail to ensure it is enabled and logging events.
+
+   ```sh
+   aws cloudtrail get-trail-status --name my-trail
+   ```
+
+4. **Ensure Lambda Events are Included:**
+   By default, CloudTrail logs all management events, including those from AWS Lambda. However, you can explicitly ensure that Lambda events are included by checking the event selectors.
+
+   ```sh
+   aws cloudtrail put-event-selectors --trail-name my-trail --event-selectors '[{"ReadWriteType": "All", "IncludeManagementEvents": true}]'
+   ```
+
+By following these steps, you can ensure that CloudTrail is enabled and properly configured to log AWS Lambda events using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To ensure that AWS CloudTrail is enabled for AWS Lambda using Python scripts, you can follow these steps:
+
+1. **Install Boto3 Library**:
+   Ensure you have the Boto3 library installed, which is the AWS SDK for Python. You can install it using pip if you haven't already.
+
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session and create a CloudTrail client. This will allow you to interact with the CloudTrail service.
+
+   ```python
+   import boto3
+
+   # Initialize a session using Amazon EC2
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+
+   # Create CloudTrail client
+   cloudtrail_client = session.client('cloudtrail')
+   ```
+
+3. **Check if CloudTrail is Enabled**:
+   Write a function to check if CloudTrail is enabled. This function will list all trails and check their status.
+
+   ```python
+   def is_cloudtrail_enabled():
+       trails = cloudtrail_client.describe_trails()
+       for trail in trails['trailList']:
+           status = cloudtrail_client.get_trail_status(Name=trail['Name'])
+           if status['IsLogging']:
+               return True
+       return False
+   ```
+
+4. **Enable CloudTrail if Not Enabled**:
+   If CloudTrail is not enabled, create a new trail and start logging.
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>